### PR TITLE
Sweets/8/8

### PR DESCRIPTION
**Pull request by sweetmantech**

### CODE REVIEW

File: lib/collectionHolder.ts [hunk 1]
```diff
@@ -109,20 +109,18 @@ export const getAvailableFreeMints = async (
...
   const contract = new ethers.Contract(
     process.env.NEXT_PUBLIC_COLLECTION_HOLDER,
     collectionHolderAbi,
     signer,
   )
   try {
-    const address = await signer.getAddress()
     const tx = await contract.mint(
-      passportIds,
+      [passportIds[passportIds.length - 1]],
```
Comment (crazzywizard) [inline lib/collectionHolder.ts]: what are you trying to do here?

File: lib/merkle/generateMerkleProof.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import MerkleTree from "merkletreejs"
```
Comment (crazzywizard) [inline lib/merkle/generateMerkleProof.ts:1]: we can prob move this to api, make it easier to get root and proof. I am assuming you would have to do this every time someone mints on cre8orlist day. 
make an api that takes a param of address and move all of this generation to server side.

File: components/MintPage/MintDay/MintBoard/MintBoard.tsx [hunk 3]
```diff
@@ -1,71 +1,45 @@
...
-      case 2:
-        if (!tierIIQuantity) return
-        setTierIIQuantity(tierIIQuantity - 1)
-        break
-      case 3:
-        if (!tierIIIQuantity) return
-        setTierIIIQuantity(tierIIIQuantity - 1)
-        break
-      default:
-    }
```
Comment (crazzywizard) [inline components/MintPage/MintDay/MintBoard/MintBoard.tsx:68]: interesting. I went with a different way initially. I left this code as is and created an updateCart function that sets the carts tiers and quantity on button click.

File: components/MintPage/MintDay/Modals/MintMoreModal.tsx [hunk 4]
```diff
@@ -10,26 +10,13 @@ import { useMintProvider } from "../../../../providers/MintProvider"
...
-  coreMintFunc,
-  handleLoading,
-  openSuccessModal,
-}) => {
+const MintMoreModal: FC<MintMoreModalProps> = ({ isModalVisible, toggleIsVisible, loading }) => {
   const isXl = useMediaQuery("(max-width: 1150px)")
 
-  const { checkNetwork, refetchInformation, leftQuantityCount, lockedCntOfCre8or } =
-    useMintProvider()
+  const { leftQuantityCount, lockedCntOfCre8or } = useMintProvider()
```
Comment (crazzywizard) [inline components/MintPage/MintDay/Modals/MintMoreModal.tsx:16]: you need locked and unlocked id's correct? I am going to add that logic to initalData retrieval using multicall. Just having the number is useless. We need to know if there are any unlocked.

**Review by crazzywizard — APPROVED** [review]
Minor Comments 
LGTM!

File: hooks/useMintCart.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { useState } from "react"
```
Comment (crazzywizard) [inline hooks/useMintCart.tsx:1]: for future we could have prob used a `useReducer` hook for this.

File: lib/collectionHolder.ts (hunk 1 above)
Comment (sweetmantech) [reply lib/collectionHolder.ts]: fixed with your tip to use `availablePassportIds` ✅ 

Screenshot 2023-08-01 at 9 08 43 PM

File: lib/merkle/generateMerkleProof.ts (hunk 2 above)
Comment (sweetmantech) [reply lib/merkle/generateMerkleProof.ts:1]: migrated merkle proof generation to `/api/merkle?address={X}` ✅

**Pull request merged**